modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,420 @@
-        if (query1)
-          query1.answer.each do |ip|
-            print_status("Start of Authority: #{rr.mname} IP address: #{ip.address} Record: SOA")
-            report_note(:host => @nsinuse.to_s,
-              :proto => 'udp',
-              :sname => 'dns',
-              :port => 53 ,
-              :type => 'dns.enum',
-              :update => :unique_data,
-              :data => "#{ip.address.to_s},#{rr.mname},SOA")
-          end
-        end
-      end
+  def dns_bruteforce(domain)
+    wordlist = datastore['WORDLIST'].to_s
+    return if wordlist.blank?
+
+    threadnm = datastore['THREADS'].to_i
+    return if threadnm == 0
+
+    queue = ::Queue.new
+    File.foreach(wordlist) do |line|
+      queue << "#{line.chomp}.#{domain}"
     end
-    query = @res.query(target, "NS")
-    if (query)
-      (query.answer.select { |i| i.class == Net::DNS::RR::NS}).each do |rr|
-        query1 = @res.search(rr.nsdname)
-        if (query1)
-          query1.answer.each do |ip|
-            next unless ip.class == Net::DNS::RR::A
-            print_status("Name Server: #{rr.nsdname} IP address: #{ip.address} Record: NS")
-            report_note(:host => @nsinuse.to_s,
-              :proto => 'udp',
-              :sname => 'dns',
-              :port => 53 ,
-              :type => 'dns.enum',
-              :update => :unique_data,
-              :data => "#{ip.address.to_s},#{rr.nsdname},NS")
+
+    records = []
+    until queue.empty?
+      tl = []
+      1.upto(threadnm) do
+        tl << framework.threads.spawn(
+          "Module(#{refname})-#{domain}", false, queue.shift) do |target|
+            Thread.current.kill unless target
+            a = get_a(target)
+            records |= a if a
           end
-        end
-      end
-    end
-    query = @res.query(target, "MX")
-    if (query)
-      (query.answer.select { |i| i.class == Net::DNS::RR::MX}).each do |rr|
-        print_status("Name: #{rr.exchange} Preference: #{rr.preference} Record: MX")
-        report_note(:host => @nsinuse.to_s,
-          :proto => 'udp',
-          :sname => 'dns',
-          :port => 53 ,
-          :type => 'dns.enum',
-          :update => :unique_data,
-          :data => "#{rr.exchange},MX")
-      end
-    end
-    query = @res.query(target, "TXT")
-    if (query)
-      query.answer.each do |rr|
-        print_status(rr.inspect)
-        print_status("Text: #{rr.inspect}")
-        report_note(:host => @nsinuse.to_s,
-          :proto => 'udp',
-          :sname => 'dns',
-          :port => 53 ,
-          :type => 'dns.enum',
-          :update => :unique_data,
-          :data => rr.inspect)
       end
+      break if tl.length == 0
+      tl.first.join
+      tl.delete_if { |t| !t.alive? }
     end
+    records
   end
 
-  def tldexpnd(targetdom,nssrv)
-    target = targetdom.scan(/(\S*)[.]\w*\z/).join
-    target.chomp!
-    if not nssrv.nil?
-      @res.nameserver=(nssrv)
-      @nsinuse = nssrv
+  def dns_wildcard(domain)
```

Comment:
This looks to be designed to return a boolean indicating whether or not DNS wildcards appear to be in use, so please name the method accordingly -- `dns_wildcard_enabled?(domain)`